### PR TITLE
[stable/redis-ha] Consistent label "name" for all deployments (Issue 4517)

### LINKED ISSUE #4517: [stable/redis-ha] Inconsistency in deployment label names
FEATURE REQUEST

**Which chart**:
Chart: redis-ha

**Description**:

The server and sentinel deployment uses different name (labels).

For sentinel:
`spec.template.metadata.labels.name: {{ template "fullname" . }}-sentinel`

For server:
`spec.template.metadata.labels.name: "redis-sentinel"`

They should probably be the same format for consistency. They should be either:
`{{ template "fullname" . }}-sentinel` and `{{ template "fullname" . }}-server`
or both should be:
`redis-sentinel` and `redis-server`.

### PR DESCRIPTION
**Pull request by roychri**
**What this PR does / why we need it**:

Make the label "name" of the deployments consistent.

**Which issue this PR fixes**:

fixes #4517

**Special notes for your reviewer**:

I didn't know if I should have bumped the version or not or if that is done when you pull, I am new to this contributing stuff. Let me know how I can improve things.

@smileisak  @ianmaddox

### CODE REVIEW

**Comment by unguiculus** [comment]
I'd prefer it if you update the chart to have standard labels. See review guidelines.

https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

**Comment by roychri** [comment]
@unguiculus Standard labels are already being addressed by the template `labels.standard`.
However, I see that `app` is defined both in the deploy file and then in the template `labels.standard` and that it uses a non standard labels like `name` and `redis-node`.

Do you mean that I should do one or more of the following?
* Remove the dupe `app` label
* Use the name instead of fullname for the `app` label
* Remove the non standard `name` label
* Add the `component` label (since we are using 2 components)

Or are you referring to something else like not using a template for the standard labels?

Edit: It is important to mention that `redis-node` label is used by a script in the docker image which I have no control to, so to remove that label would require someone to also modify that docker image. This means that I would recommend to remove that `redis-node` label in another PR if that's possible.

Thanks.

**Comment by unguiculus** [comment]
Sorry, I overlooked that standard labels are set. Dupes should be removed, the `app` label in `standard.labels` is correct. Adding components would be a good thing as well as namespacing. Compare with what `helm create` using Helm 2.8+ give you.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Test pods should also have standard labels.

**Comment by roychri** [comment]
@unguiculus oops, my bad. I followed this and they do not use standard labels on this page.
Maybe I should do a PR to include standard labels on that page too! :)
https://github.com/kubernetes/helm/blob/master/docs/chart_tests.md
I will update my PR as soon as I figure out a way to fix my local and remote repo which I think I screwed up when trying to put the previous change. I am new to this fork/contrib stuff...

**Comment by roychri** [comment]
@unguiculus Sorry about all the activity, I fixed my local repo and pushed the changes for the standard labels in the tests too.

**Comment by roychri** [comment]
@unguiculus Are my changes appropriate for what you were expecting?

File: stable/redis-ha/templates/tests/test-redis-master-service.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-master-service-test"
```
Comment (unguiculus) [inline stable/redis-ha/templates/tests/test-redis-master-service.yaml]: {{ template "redis-ha.fullname" . }}-master-service-test

File: stable/redis-ha/templates/tests/test-redis-sentinel-service.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-sentinel-service-test"
```
Comment (unguiculus) [inline stable/redis-ha/templates/tests/test-redis-sentinel-service.yaml]: {{ template "redis-ha.fullname" . }}-sentinel-service-test

File: stable/redis-ha/templates/tests/test-redis-slave-service.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-slave-service-test"
```
Comment (unguiculus) [inline stable/redis-ha/templates/tests/test-redis-slave-service.yaml]: {{ template "redis-ha.fullname" . }}-slave-service-test

**Review by unguiculus — CHANGES_REQUESTED** [review]
Sorry for the delay.

File: stable/redis-ha/templates/tests/test-redis-master-service.yaml (hunk 1 above)
Comment (roychri) [reply stable/redis-ha/templates/tests/test-redis-master-service.yaml]: Changed

File: stable/redis-ha/templates/tests/test-redis-sentinel-service.yaml (hunk 2 above)
Comment (roychri) [reply stable/redis-ha/templates/tests/test-redis-sentinel-service.yaml]: Changed

File: stable/redis-ha/templates/tests/test-redis-slave-service.yaml (hunk 3 above)
Comment (roychri) [reply stable/redis-ha/templates/tests/test-redis-slave-service.yaml]: Changed

**Pull request merged**